caikit/core/model_management/multi_model_initializer.py

```diff
@@ -0,0 +1,146 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+The MultiModelInitializer configures a set of other model initializers that will be used
+in sequence to try loading models.
+
+Configuration for MultiModelInitializer lives under the config as follows:
+
+model_management:
+    initializers:
+        <initializer name>:
+            type: MULTI
+            config:
+                # Sequence of other initializer names to use in priority order
+                initializer_priority:
+                    - other_initializer1
+                    - other_initializer2
+"""
+# Standard
+from typing import Optional
+
+# First Party
+import aconfig
+import alog
+
+# Local
+from ...config import get_config
+from ..exceptions import error_handler
+from ..modules import ModuleBase, ModuleConfig
+from .model_initializer_base import ModelInitializerBase
+
+# NOTE: Top-level import done so that global MODEL_MANAGER can be used at
+#   construction time without incurring a circular dependency
+import caikit.core
+
+log = alog.use_channel("MINIT")
+error = error_handler.get(log)
+
+
+class MultiModelInitializer(ModelInitializerBase):
+    __doc__ = __doc__
+
+    name = "MULTI"
+
+    def __init__(self, config: aconfig.Config, instance_name: str):
+        """Initialize with the sequence of initializers to use"""
+        self._instance_name = instance_name
+        initializer_priority = config.initializer_priority
+        error.type_check(
+            "<COR47518221E>",
+            list,
+            initializer_priority=initializer_priority,
+        )
+        error.type_check_all(
+            "<COR47518222E>",
+            str,
+            initializer_priority=initializer_priority,
+        )
+        error.value_check(
+            "<COR05042343E>",
+            initializer_priority,
+            "Must provide at least one valid initializer",
+        )
+        config_initializers = get_config().model_management.initializers
+        invalid_initializers = [
+            initializer
+            for initializer in initializer_priority
+            if initializer not in config_initializers
+        ]
+        error.value_check(
+            "<COR68252034E>",
+            not invalid_initializers,
+            "Invalid initializers given in initializer_priority: {}",
+            invalid_initializers,
+        )
+        error.value_check(
+            "<COR54613971E>",
+            self._instance_name not in config_initializers,
+            "Cannot include self in multi initializer priority",
+        )
+        model_manager = config.model_manager or caikit.core.MODEL_MANAGER
+        log.debug2(
+            "Setting up %s with initializer priority: %s",
+            self.name,
+            initializer_priority,
+        )
+        self._initializers = [
```

Comment:
I love how many validation lines it takes to get to this single comprehension! No one is getting through this with a bad config...